### ISSUE TITLE
A test BaseBufferTest throws when calling compact() in the JVM.

### ISSUE THREAD

**Issue by nicolasstucki** [scala-js/scala-js#2192]

**Comment by sjrd** [comment]
This is a bug in the JDK. It was fixed in JDK 9. The following Java source file reproduces the bug on JDK 8 but not on 9, 10 nor 11 EA:
```java
import java.nio.*;

public class Test {
    public static void main(String[] args) {
        ByteBuffer buf = ByteBuffer.allocate(65);
        buf.position(17);
        buf.limit(17 + 40);

        ByteBuffer slice = buf.slice();
        slice.order(ByteOrder.LITTLE_ENDIAN);

        IntBuffer view = slice.asIntBuffer();

        assert view.capacity() == 10; // 40 / 4
        assert view.position() == 0;
        assert view.limit() == 10;

        view.put(new int[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 });
        view.position(0);
        assert view.remaining() == 10;

        view.position(6);
        view.compact();

        assert view.position() == 4;
        assert view.limit() == 10;
        for (int i = 0; i < 4; i++)
          assert view.get(i) == i + 6;

        System.out.println("done");
    }
}
```
So our implementation is correct. We should at some point enable the test on the JVM is the version is >= 9.

**Comment by gzm0** [comment]
Fixed in 07c2c6e07b391858c18cc782bebbc33b035a2f02
